local-cli/runAndroid/runAndroid.js

```diff
@@ -78,13 +78,101 @@ function tryRunAdbReverse(device) {
 // Builds the app and runs it on a connected emulator / device.
 function buildAndRun(args) {
   process.chdir(path.join(args.root, 'android'));
+  const cmd = process.platform.startsWith('win')
+    ? 'gradlew.bat'
+    : './gradlew';
+
+  const packageName = fs.readFileSync(
+      'app/src/main/AndroidManifest.xml',
+      'utf8'
+    ).match(/package="(.+?)"/)[1];
+
+  const adbPath = getAdbPath();
+  if (args.deviceId) {
+    runOnSpecificDevice(args, cmd, packageName, adbPath);
+  } else {
+    runOnAllDevices(args, cmd, packageName, adbPath);
+  }
+}
+
+function runOnSpecificDevice(args, gradlew, packageName, adbPath) {
+  let devices = adb.getDevices();
+  let selectedDevice = devices.filter(function(deviceId) {
+    return ((deviceId === args.deviceId) ? deviceId : false);
+  });
+  if (selectedDevice.length > 0) {
```

Comment:
Could be `if (devices.indexOf(args.deviceId) !== -1)` using `args.deviceId` in the call below, and then there's no need for the filter function. The call won't work as-is because `selectedDevice` is actually an array.